src/mscorlib/src/System/IO/StreamReader.cs

```diff
@@ -71,6 +71,9 @@ internal static int DefaultBufferSize
         // This is used only for preamble detection
         private int bytePos;
 
+        [NonSerialized]
+        private StringBuilder builder;
```

Comment:
I can't figure out how to add the comment in the right place, but you should null out `builder` in the Dispose method around line 276.